src/lsp_data.rs

```diff
@@ -80,6 +87,16 @@ pub mod ls_util {
         }
     }
 
+    pub fn location_from_racer_match(mtch: racer::Match) -> Option<Location> {
+        let source_path = &mtch.filepath;
+
+        if let Some(racer::Coordinate { line, column }) = mtch.coords {
+            Some(location_from_position(source_path, line - 1, column))
+        } else {
+            None
+        }
+    }
+
```

Comment:
Nope: the ls_types crate is meant to be independent of which language the LS analyses. (Indeed @marwes uses it for https://github.com/gluon-lang/gluon for example). Adding a Racer dependency would not be adequate at all, because Racer is an engine for Rust code.

If anything, we could create an smallish `ls-commons` crate with code that would be of use for any language engine. (stuff like positions, ranges, mapping from offset-based positions to column/line etc. - I imagine even more stuff might be suitable to be added to an `ls-commons` crate in the future). Then `ls_types` could use the `ls-commons` crate.

I was actually gonna ask if `rls-vfs` could be abstracted a bit better so as to be independent of `rls-analysis`, and anything else Rust specific. I actually wanted to use `rls-vfs` to build a simple LS for an example language, and although I can see now I won't have free time to do that (changing jobs), it doesn't change the fact that it would be a better, more general design for the `rls-vfs` crates.